uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/shared/components/leave-request-actions.component.js

```diff
@@ -116,13 +120,31 @@ define([
     };
 
     vm.allowedActions = [];
+    vm.loading = { component: true };
 
     vm.action = action;
     vm.openLeavePopup = openLeavePopup;
 
     (function init () {
-      indexSupportData();
-      setAllowedActions();
+      $q.resolve()
+        .then(indexSupportData)
```

Comment:
Does not look like `indexSupportData` and `loadCurrentlyLoggedInContactId` are dependent on each other, so they should be called in parallel and not in series in order to optimize performance.

Reply:
`indexSupportData` is a sync func. If you run it "in parallel" the `indexSupportData` will still be executed before `loadCurrentlyLoggedInContactId`. So I just kept it for readability.